daemon/src/maker_cfd.rs

```diff
@@ -516,47 +555,57 @@ impl Actor {
 
     async fn handle_accept_settlement(&mut self, order_id: OrderId) -> Result<()> {
         tracing::debug!(%order_id, "Maker accepts a settlement proposal" );
-        // TODO: Initiate the settlement
 
-        self.current_pending_proposals
-            .remove(&order_id)
-            .context("Could not find proposal for given order id")?;
-        self.send_pending_proposals()?;
+        let taker_id = self.get_taker_id_of_proposal(&order_id)?;
+
+        // TODO: Initiate the settlement - should we start calculating the
+        // signature here?
```

Comment:
I don't think so. The maker can reactively do this upon receiving the signature from the taker. I would do that in a completely different message and simply assert that upon receiving that message, we have actually agreed to a settlement.